books_to_scrape/spiders/books.py

```diff
@@ -0,0 +1,53 @@
+import re
+from pathlib import Path
+from types import new_class
+
+import scrapy
+from scrapy import Selector
+from scrapy.http import Response
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/index.html"]
+
+    def parse(self, response: Response, **kwargs):
+        book_detail_links = response.css("h3 > a::attr(href)").getall()
+
+        yield from response.follow_all(book_detail_links, self._parse_book)
+
+        next_page = response.css("li.next > a::attr(href)").get()
+
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def _parse_book(self, response: Response, **kwargs) -> float:
```

Comment:
The return type annotation `-> float` for the `_parse_book` method is incorrect as the method yields a dictionary, not a float. It should be updated to `-> None` or removed.